services/settings/js/app.js

```diff
@@ -92,21 +92,42 @@
     // * lock.set || lock.get
     // All the operations have a requestId, and the lock operations also include
     // a lock id.
-    var request = evt.data.remoteData;
+    var request = aEvt.data.remoteData;
     var requestOp = request.data.operation;
+    var targetURL = aEvt.data.targetURL;
+
+    var forbidCall = function(constraints) {
+      var settings = [];
+      switch(requestOp) {
+        case 'addObserver':
+        case 'removeObserver':
+          settings = [request.data.params.settingName];
```

Comment:
Isn't params an array always? shouldn't this be params[0].settingName? (or just params[0])